model/patch/github.go

```diff
@@ -155,24 +158,39 @@ func NewGithubIntent(msgDeliveryID, patchOwner, calledBy string, pr *github.Pull
 		patchOwner = pr.User.GetLogin()
 	}
 
+	// get the patchId to repeat the definitions from
+	repeat := getRepeatPatchId(pr.Number)
+
 	return &githubIntent{
-		DocumentID:   msgDeliveryID,
-		MsgID:        msgDeliveryID,
-		BaseRepoName: pr.Base.Repo.GetFullName(),
-		BaseBranch:   pr.Base.GetRef(),
-		HeadRepoName: pr.Head.Repo.GetFullName(),
-		PRNumber:     pr.GetNumber(),
-		User:         patchOwner,
-		UID:          int(pr.User.GetID()),
-		HeadHash:     pr.Head.GetSHA(),
-		BaseHash:     pr.Base.GetSHA(),
-		Title:        pr.GetTitle(),
-		IntentType:   GithubIntentType,
-		PushedAt:     pr.Head.Repo.PushedAt.Time.UTC(),
-		CalledBy:     calledBy,
+		DocumentID:    msgDeliveryID,
+		MsgID:         msgDeliveryID,
+		BaseRepoName:  pr.Base.Repo.GetFullName(),
+		BaseBranch:    pr.Base.GetRef(),
+		HeadRepoName:  pr.Head.Repo.GetFullName(),
+		PRNumber:      pr.GetNumber(),
+		User:          patchOwner,
+		UID:           int(pr.User.GetID()),
+		HeadHash:      pr.Head.GetSHA(),
+		BaseHash:      pr.Base.GetSHA(),
+		Title:         pr.GetTitle(),
+		IntentType:    GithubIntentType,
+		PushedAt:      pr.Head.Repo.PushedAt.Time.UTC(),
+		CalledBy:      calledBy,
+		RepeatPatchId: repeat,
 	}, nil
 }
 
+// getRepeatPatchId returns the patch id to repeat the definitions from
+// this information is found on the most recent pr patch
+func getRepeatPatchId(prNumber *int) string {
+	p, err := FindOne(MostRecentPatchByPR(*prNumber))
+	if err != nil || p == nil {
```

Comment:
Seems we should still be aware of errors that happen here. Can we just return an empty string if `p == nil` and return the error otherwise?